src/scripts/sync_subclassof.py

```diff
@@ -0,0 +1,387 @@
+    # - rels_mondo_mondo: Edges with Mondo IDs that appear in Mondo, derived from all source::Mondo mappings
+    # - rels_mondo_source: Edges with Mondo IDs that appear in the source, derived from all source::Mondo mappings
+    rels_mondo_mondo = set([x for x in rels_mondo_raw if x[2].startswith('MONDO')])  # filter out non-source parents
+    rels_mondo_source = set()
+    # todo: 2 line alternative: will only work if mondo.db and mondo.sssom.tsv are in sync @joe's-new-computer-needed
+    #  - these two lines will replace the for loop with the try/excepts & following two sets
+    #  - temp: rels_mondo_non_source & rels_mondo_non_synced are temporary while these 2 files out of sync
+    #  - perhaps it will be useful to keep the 'non synced check' above and throw an error? but probably not
+    # rels_mondo_source = [(mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]) for x in rels_mondo_mondo]
+    # rels_mondo_source = [x for x in rels_mondo_source if any([x[0].startswith(y) for y in owned_prefix_map]) and
+    #   any([x[2].startswith(y) for y in owned_prefix_map])]
+    rels_mondo_non_source_temp = set()
+    rels_mondo_non_synced_temp = set()
+    # - rels_mondo_non_source_temp: Parent and/or child not from source
+    # - rels_mondo_non_synced_temp: Parent and/or child Mondo ID exists in mondo.db but not mondo.sssom.tsv
+    for x in rels_mondo_mondo:
+        try:
+            rels_mondo_source.add((mondo_source_map[x[0]], x[1], mondo_source_map[x[2]]))
+        except KeyError:
+            try:
+                rels_mondo_non_source_temp.add((mondo_all_sources_map[x[0]], x[1], mondo_all_sources_map[x[2]]))
+            except KeyError:
+                rels_mondo_non_synced_temp.add(x)
+    t1 = datetime.now()
+    print(f'  Completed in {(t1 - t0).seconds} seconds')
+
+    # Determine hierarchy diferences -----------------------------------------------------------------------------------
+    # - find which edges appear in both Mondo and source, or only in one or the other
+    in_both_source_ids_set: Set[RELATIONSHIP] = rels_source.intersection(rels_mondo_source)
+    in_source_only_source_ids_set: Set[RELATIONSHIP] = rels_source.difference(rels_mondo_source)
+    in_mondo_only_source_ids_set: Set[RELATIONSHIP] = rels_mondo_source.difference(rels_source)
+
+    # Create outputs ---------------------------------------------------------------------------------------------------
+    robot_subheader = [{
+        'subject_mondo_id': 'ID',
+        'subject_mondo_label': '',
+        'object_mondo_id': 'SC %',
+        'subject_source_id': '>A oboInOwl:source',
+        'object_source_id': '',
+        'object_mondo_label': '',
+    }]
+    robot_subheader_obs = deepcopy(robot_subheader)
+    # todo: is this correct, or do we need to change or add something new like MONDO:excludeSubClassRelationship
+    robot_subheader_obs[0]['object_mondo_id'] = 'MONDO:excludeObsoleteSource'
```

Comment:
## `SOURCE.subclass.added-obsolete.tsv` robot val for `object_mondo_id` @matentzn 
@matentzn Let me know if I should change the value I assigned.
<details><summary>Details</summary>
<p>

@matentzn At last meeting, we decided to split this table into 2:

1. `SOURCE.subclass.added.tsv`
2. `SOURCE.subclass.added-obsolete.tsv`

For (2), I also needed to change the robot subheader value for `object_mondo_id` from `SC %` to something else. Off the top of your head, you said it might be something perhaps like `MONDO:excludeSubClassRelationship`, but that does not exist. I checked [exclude list in the docs](https://mondo.readthedocs.io/en/latest/editors-guide/exclusion-reasons/), and also the `mondo` repo just in case the docs were up to date, but there's nothing like that there. I couldn't find anything that really looked good, except maybe `MONDO:excludeObsoleteSource`. However, these terms are not necessarily obsolete in the source; they are obsolete in Mondo.

I'm using `MONDO:excludeObsoleteSource` for now, but let me know if you advise different.

</p>
</details>

Reply:
- [ ] `http://purl.obolibrary.org/obo/mondo#excluded_subClassOf`